pkg/gits/operations/pull_request_op_test.go

```diff
@@ -79,30 +79,83 @@ func TestCreatePullRequest(t *testing.T) {
 	err := testhelpers.CreateTestEnvironmentDir(o.CommonOptions)
 	assert.NoError(t, err)
 
-	o.GitURLs = []string{"testowner/testrepo"}
-	o.SrcGitURL = "testowner/testrepo"
-	o.Version = "3.0.0"
-
 	pegomock.When(gitter.HasChanges(pegomock.AnyString())).ThenReturn(true, nil)
 
+	return o
+}
+
+func TestCreatePullRequest(t *testing.T) {
+	prOpts := setupTestPullRequestOperation(t)
+
+	prOpts.GitURLs = []string{"testowner/testrepo"}
+	prOpts.SrcGitURL = "testowner/testrepo"
+	prOpts.Version = "3.0.0"
+
 	var results *gits.PullRequestInfo
+	var err error
 
 	logOutput := log.CaptureOutput(func() {
-		results, err = o.CreatePullRequest("test", func(dir string, gitInfo *gits.GitRepository) (strings []string, e error) {
+		results, err = prOpts.CreatePullRequest("test", func(dir string, gitInfo *gits.GitRepository) (strings []string, e error) {
 			return []string{"1.0.0", "v1.0.1", "2.0.0"}, nil
 		})
 		assert.NoError(t, err)
-		assert.NotNil(t, results)
+		assert.NotNil(t, results, "we must have results coming out of the PR creation")
 	})
 
 	assert.Contains(t, logOutput, "Added label updatebot to Pull Request https://fake.git/testowner/testrepo/pulls/1",
 		"Updatebot label should be added to the PR")
 
-	assert.NotNil(t, results, "we must have results coming out of the PR creation")
 	assert.Equal(t, "chore(deps): bump testowner/testrepo from 1.0.0, 2.0.0 and v1.0.1 to 3.0.0",
 		results.PullRequestArguments.Title, "The PR title should contain the old and new versions")
 }
 
+func TestCreatePullRequestsWithLabels(t *testing.T) {
+	prOpts := setupTestPullRequestOperation(t)
+
+	prOpts.GitURLs = []string{"testowner/testrepo"}
+	prOpts.SrcGitURL = "testowner/testrepo"
+	prOpts.Version = "2.0.0"
+	fromVersion := "1.0.0"
+
+	results, err := prOpts.CreatePullRequest("test", func(dir string, gitInfo *gits.GitRepository) (strings []string, e error) {
+		return []string{fromVersion}, nil
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, results, "we must have results coming out of the PR creation")
+
+	prNumber := 1
+	assert.Equal(t, prNumber, *results.PullRequest.Number,"This should be PR number %d", prNumber)
+
+	assert.Equal(t, fmt.Sprintf("chore(deps): bump %s from %s to %s", prOpts.SrcGitURL, fromVersion, prOpts.Version),
+		results.PullRequestArguments.Title, "The PR title should contain the old and new versions")
+
+	assert.Equal(t, 1, len(results.PullRequest.Labels),"One label expected")
```

Comment:
Look at that, we already have access to the labels via the returned pull request details :-) 👍

Reply:
What's about adding an assert for explicitly 'update-bot'. This kind of codifies the label we expect here